### ISSUE TITLE
Terrain glitches badly in MPE

### ISSUE THREAD

**Issue by ra4nd0m** [prestja/Kopernicus#26]
![image](https://user-images.githubusercontent.com/58385866/88467631-915d5d80-cee1-11ea-9018-c21a3cbd942f.png)
The ground flickers very fast and causes lag. On global time warp the ground moves with the ship. This happens on non-spherical bodies.

**Comment by R-T-B** [comment]
Interesting.  Thank you for the report.  For reproducing and debuging, can you name a good planet pack and planet to teleport too to test this?

MPE = Minor Planet's expansion, I assume?  Which body is that?

**Comment by hemeac** [comment]
[KSP Log Lint-Mikey.zip](https://github.com/prestja/Kopernicus/files/4977044/KSP.Log.Lint-Mikey.zip)

As stated in the forum, this is Lint-Mikey from MPE.  Using Hyperedit and then reloading from a quicksave, I was not able to reproduce the issues facing ra4nd0m.  Tested with 1.9.1 and Kopernicus release, 1.9.1-6

**Comment by R-T-B** [comment]
For right now, I'm going to assume another mod is interfering or you aren't on latest release.  If someone can reproduce this issue reliably, I will reopen the issue, of course.

**Comment by ra4nd0m** [comment]
I've done a bit more testing on this topic. When you start approaching the object at certain point the lag just disappears. If you leave SOI of an object the lag disappears. But after returning into SOI at a height of 10000m the lag returns. If you again try to land the lag disappears again. But when you reach height of about 10000m it starts lagging again. If there was some kind of Sphere of Lag around an object. 
[Player.log](https://github.com/prestja/Kopernicus/files/4977335/Player.log)
The log is attached. KSP install is 100% pure.
![image](https://user-images.githubusercontent.com/58385866/88475279-0f981f00-cf37-11ea-9f73-6386b77fa619.png)

**Comment by hemeac** [comment]
@ra4nd0m, I'll check to see if I can reproduce under those conditions in a couple hours and report back to see if I notice anything similar, I just checked previously within a low orbit.

**Comment by hemeac** [comment]
Under those conditions, I was able to reproduce @ra4nd0m issues even in a barebones install.
[Lint-Mikey2.zip](https://github.com/prestja/Kopernicus/files/4977558/Lint-Mikey2.zip)

Here's a video that shows the issue: https://www.youtube.com/watch?v=SOdv0dtpESg

**Comment by R-T-B** [comment]
Thank you for the additional data, reopening.

Could someone see if this occurs in 1.8 as well?

**Comment by democat3457** [comment]
After peering over the configs, it could be that the ScaledVersion fades out at about 9,000m, whereas the PQS fades in at 135,000m. Since the PQS fades in over a distance of 35,000m, there is little lag; however, the ScaledVersion practically vanishes immediately as it fades out over a distance of 100m. This immediate render change could be causing said lag spike at ~10,000m.

**Comment by democat3457** [comment]
Also, the ScaledVersion `fadeStart` is greater than the `fadeEnd`, which is likely responsible - the values are supposed to be reversed, otherwise the ScaledVersion will have an increasing gradient from 
```
-------          /-         ScaledVersion present
           /----/
        -/          -----   ScaledVersion not present
```

**Comment by R-T-B** [comment]
Yep, if it occurs in 1.8 as well then I'm going to mark this an author/config issue.

**Comment by hemeac** [comment]
I think I know just enough about editing configuration files to test that.  Will do that after work and also try a 1.8 test.

**Comment by hemeac** [comment]
This issue appears to have been an issue in 1.8 as well, so this is not a bug introduced in the 1.9 builds.  However, it doesn't seem to be isolated to MPE.  I found that RAB-58E from the Grannus Expansion Pack exhibits similar behavior and didn't have the issue with fading swapped that @democat3457 found in the configuration file of Lint-Mikey.  RAB is also a "tiny" planet body so perhaps that may be related as this is an issue that I didn't have with other bodies in the planet packs, but admittedly did not try them all. Log file attached as well as showing RAB-58E video (HD quality still processing at the time of posting, so you may want to wait if you want to be able to see more than blurry video): https://youtu.be/Ur_eQHGy16c
[KSP.zip](https://github.com/prestja/Kopernicus/files/4981277/KSP.zip)

**Comment by democat3457** [comment]
Hmm the only thing I can find in the configs related to 69,000m (the approx. height of the lag) is the PQS deactivateAltitude, which is 70k; however, Lint-Mikey's deactivateAltitude is 135,000m, which doesn't seem to be close to the lag location...

**Comment by ra4nd0m** [comment]
I assume that I can rename this issue as wierd lag and glitches on small non-spherical boides?) By the way none of the OPM non-spherical boides have shown such behavior. 
If this issue was in 1.8 I assume that kopernicus soon be published in CKAN?

**Comment by R-T-B** [comment]
Yep, we are pretty close to CKAN.  I have a feeling this is a pack-bug/not our fault, but a 1.8 test would confirm...

EDIT:  Oh darn.  It looks like an old bug but still a bug if it occurs in a pack without config issues.  Yes, please alter bug title to fit.  It may be a stock bug we can't work around, but let's hope not.

  The only other stopper bug to CKAN is the solar flux/EC calc one, and I have a fix actively being tested for that in bleeding edge.

We aren't far from CKAN at all.  And yes a rename to

**Comment by R-T-B** [comment]
> If this issue was in 1.8 I assume that kopernicus soon be published in CKAN?

At this time, I do not consider this issue a blocker to CKAN.  Comment if you disagree, but reasons are:

1.) I don't have any idea how to fix it right now, so we'd be waiting forever

2.)  It's pretty minor for most users and very situational.

Critique of this welcome.

**Comment by hemeac** [comment]
I would agree.  Given that it doesn't occur at the surface, it wouldn't impact any critical aspects of the gameplay like landing.
